src/main/java/com/exacaster/deltafetch/rest/ApiController.java

```diff
@@ -16,6 +17,7 @@ public ApiController(Router router) {
     }
 
     @Get(value = "/{+path*}", consumes = MediaType.ALL)
+    @ExecuteOn("api")
```

Comment:
Why not `@ExecuteOn(TaskExecutors.BLOCKING)`, to use virtual threads and avoid limited thread pool?

https://docs.micronaut.io/latest/guide/#virtualThreads

Reply:
Unfortunately, we are on Java 11 here https://github.com/exacaster/delta-fetch/blob/main/build.gradle#L15. And I did not want to adjust any internal executor services since I am not sure what else I might be impacting. Thought that safe way is to go with a custom thread pool.

Reply:
Why not upgrade to 21? Java 11 EOL passed.